### PR TITLE
refresh gerrit token periodically + query per project

### PR DESCRIPTION
**Pull request by krzyzacy**
things in this PR:
- make gerrit client re-auth every 10min
- gerrit api need to query changes per project
- patched up BaseRef/BaseSHA/Author (still need to patch up bootstrap)
- make gerrit comment back say it has triggered presubmit job(s)

/area prow
/hold

### CODE REVIEW

**Comment by krzyzacy** [comment]
opened https://github.com/kubernetes/test-infra/issues/7426 for checkout refs

**Comment by krzyzacy** [comment]
this is deployed and WAI, except: 
- gerrit need to support set account name
- bootstrap need to support a gerrit format of pull_refs

will address bootstrap in a follow-up PR, this PR is large enough :-)
/assign @cjwagner @BenTheElder

**Comment by krzyzacy** [comment]
/hold

hummm, looks like any events will change mtime for the gerrit PR, I'll probably need to check for last commit time

**Comment by krzyzacy** [comment]
/hold cancel
WAI now

**Comment by krzyzacy** [comment]
PTAL?

File: prow/gerrit/gerrit.go [hunk 1]
```diff
@@ -148,16 +174,14 @@ func (c *Controller) Sync() error {
...
+func (c *Controller) queryProjectChanges(proj string) ([]gerrit.ChangeInfo, error) {
+	pending := []gerrit.ChangeInfo{}
+
 	opt := &gerrit.QueryChangeOptions{}
-	for _, proj := range c.projects {
-		opt.Query = append(opt.Query, "project:"+proj)
-	}
-	opt.AdditionalFields = []string{"CURRENT_REVISION"}
+	opt.Query = append(opt.Query, "project:"+proj+"+status:open")
+	//opt.Query = append(opt.Query, )
```
Comment (cjwagner) [inline prow/gerrit/gerrit.go]: Please remove this.

Comment (krzyzacy) [reply prow/gerrit/gerrit.go]: done

**Pull request merged**